### ISSUE TITLE
Indentation: Multiple lambdas as method parameters

### ISSUE THREAD

**Issue by garricko** [checkstyle/checkstyle#3182]
For this code:
```java
import java.util.function.Supplier;

public class Sample {
  public static void main(String[] args) {
    foo(() -> {
      // seems happy with this
      return null;
    }, () -> {
      // seems to want more indent here
      return null;
    });
  }

  public static void foo(Supplier<Object> x, Supplier<Object> y) {}
}
```
I see the following violations with version 6.18:
```
[WARNING] src/main/java/Sample.java:[10] (indentation) Indentation: 'block' child have incorrect indentation level 6, expected level should be one of the following: 8, 10, 12.
[WARNING] src/main/java/Sample.java:[11] (indentation) Indentation: 'block rcurly' have incorrect indentation level 4, expected level should be one of the following: 6, 8.
```

**Comment by rnveach** [comment]
Please share configuration for indentation.

**Comment by garricko** [comment]
I was just using a subset of the Google style. Here are the settings:

```xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name = "Checker">
  <module name="TreeWalker">
    <module name="Indentation">
      <property name="basicOffset" value="2"/>
      <property name="braceAdjustment" value="0"/>
      <property name="caseIndent" value="2"/>
      <property name="throwsIndent" value="4"/>
      <property name="lineWrappingIndentation" value="4"/>
      <property name="arrayInitIndent" value="2"/>
    </module>
  </module>
</module>
```

**Comment by romani** [comment]
Line 8 need to be intended as +4.

**Comment by garricko** [comment]
That does seem to make Checkstyle happy, but it looks a little odd to me, and I can't figure out how to get either IntelliJ or Eclipse to respect that style (they both change it back to the above indentation). Is there some way to adjust the settings to allow the above code? I tried adjusting various things, but they seem to have undesirable side-effects

**Comment by romani** [comment]
Lambda is new for users, IDEs are not ideal too, eventually IDEs will have all use cases.

What format of code you personaly prefer to be looks like?

**Comment by garricko** [comment]
I prefer the style of the code above, which is fairly typical block style (if you open a block, the closing '}' aligns with the first character of the line where the opening '{' is placed). This seems to be the behavior of both IntelliJ and Eclipse.

**Comment by rnveach** [comment]
@romani 
I tested the code in Eclipse, IntelliJ, and NetBeans. They all format it as the garricko wants by default.
I even tried non-lambda code that is similar:
````
    foo(new TestClass() {
        public Object method() {
            return null;
        }
    });
````
And they were formatted all the same, which is also similar to how garricko wants.

I haven't tried all weird scenarios but if the new class or lambda is on a new line after a comma, then Eclipse wants it indented over.
````
    foo(0, //
        new TestClass() {
            public Object method() {
                return null;
            }
        });
````
If it just starts on the new line with no comma, then Eclipse doesn't want it indented:
````
    foo(//
    new TestClass() {
        public Object method() {
            return null;
        }
    });
````

**Comment by romani** [comment]
thanks a lot for investigation. 

Lets follow IDEs ways. Till checkstyle can fix indentation itself  (auto-fix), user will follow IDE as it do this boring job as formatting automatically.

**Comment by PatrickDattilio** [comment]
Is there any sort of workaround/ignore option we could use until checkstyle can handle these cases?

**Comment by rnveach** [comment]
There isn't an ignore option just for this specific case.

You can make a [suppression xml](http://checkstyle.sourceforge.net/config_filters.html#SuppressionFilter) or [suppression comment](http://checkstyle.sourceforge.net/config_filters.html#SuppressWithNearbyCommentFilter) for any errors.

**Comment by rnveach** [comment]
Using google_checks, version=8.16

````
import java.time.Instant;
import java.util.UUID;
import java.util.function.Supplier;

import lombok.extern.slf4j.Slf4j;

@Slf4j
public class LazyLog {
  public static void main(String[] args) {
    info("arg1 = {}, arg2 = {}", () -> {
      System.out.println(Instant.now().toString() + " constructing arg 1");
      return UUID.randomUUID();
    }, () -> {
      System.out.println(Instant.now().toString() + " constructing arg 2");
      return UUID.randomUUID();
    });
  }

  @SafeVarargs
  private static void info(final String message, final Supplier<Object>... suppliers) {
    if (log.isInfoEnabled()) {
      Object[] args = new Object[suppliers.length];
      for (int i = 0; i < suppliers.length; i++) {
        args[i] = suppliers[i].get();
      }
      log.info(message, args);
    }
  }
}
````
it passed the first lamda, but failed at the second.
